generator/address_enricher.hpp

```diff
@@ -0,0 +1,85 @@
+#pragma once
+#include "generator/feature_builder.hpp"
+
+#include "indexer/feature_utils.hpp"
+
+#include "geometry/tree4d.hpp"
+
+#include "coding/read_write_utils.hpp"
+
+#include <map>
+
+namespace generator
+{
+
+class AddressEnricher
+{
+  // Looks too big here? Avoiding duplicates is a thing, because we still check street name first.
+  static double constexpr kDistanceThresholdM = 20.0;
+
+public:
+  struct RawEntryBase
+  {
+    std::string m_from, m_to, m_street, m_postcode;
+    feature::InterpolType m_interpol = feature::InterpolType::None;
+
+    /// @name Used to compare house numbers by its integer value.
+    /// @{
+    static uint64_t constexpr kInvalidHN = 0;
+    static uint64_t GetIntHN(std::string const & hn);
+
+    static std::pair<uint64_t, uint64_t> constexpr kInvalidRange{kInvalidHN, kInvalidHN};
+    std::pair<uint64_t, uint64_t> GetHNRange() const;
+    /// @}
+
+    template <class TSink> void Save(TSink & sink) const
+    {
+      rw::Write(sink, m_from);
+      rw::Write(sink, m_to);
+      rw::Write(sink, m_street);
+      rw::Write(sink, m_postcode);
```

Comment:
Any chance that this sequence will change when open addresses are supported?

Reply:
What is the format used for Open Addresses? Should it be taken into an account here to avoid `if (v1) {} else { // process openaddresses }` ?